clang/include/clang/StaticAnalyzer/Core/PathSensitive/CheckerHelpers.h

```diff
@@ -110,6 +113,9 @@ class OperatorKind {
 OperatorKind operationKindFromOverloadedOperator(OverloadedOperatorKind OOK,
                                                  bool IsBinary);
 
+std::optional<DefinedSVal> getPointeeDefVal(SVal PtrSVal,
+                                            ProgramStateRef State);
```

Comment:
Perhaps consider turning this helper function into a method of `SVal`, because it's a fairly generic operation and it would be easier to find there.

Reply:
This would introduce a new dependency on `ProgramStateRef` for `SVal`, though.
And `ProgramState` already depends on `SVal`, so it makes me a bit uneasy.

Reply:
Oh, avoiding that circular dependency is a good reason to keep this function here.